geode-core/src/test/java/org/apache/geode/cache/query/internal/aggregate/AbstractAggregatorTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.query.internal.aggregate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Random;
+
+import org.junit.Test;
+
+public class AbstractAggregatorTest {
+  private final Random random = new Random();
+
+  @Test
+  public void downcastShouldReturnIntegerWhenValueIsAnIntegerAndFitsWithinTheIntegerRange() {
+    int randomInteger = random.nextInt();
+    Number integerResult = AbstractAggregator.downCast(randomInteger);
+
+    assertThat(integerResult).isInstanceOf(Number.class);
+    assertThat(integerResult).isInstanceOf(Integer.class);
+  }
+
+  @Test
+  public void downcastShouldReturnLongWhenValueIsAnIntegerButDoesNotFitsWithinTheIntegerRange() {
```

Comment:
Hi Juan , how do we ensure that the value is not in the integer range. In the "not in the Float range" unit test you did Float.MIN_VALUE - random.nextDouble(); Should we do something like that here?

Reply:
Hey @nabarunnag,

Thanks for reviewing this. The methods `Random.nextInt()`, `Random.nextLong()` and `Random.nextFloat()` always return a value within the limits of that particular `Number` type. On the other hand, `Random.nextDouble()` can return a value that fits within a `Float`, that's why I had to do some extra work for `Double`.
Hope it makes sense, let me know if you think this can be done in a better way.